docs/alloy-models/linear-fee-function/linear-fee.als

```diff
@@ -0,0 +1,313 @@
+  { e: Stutter, f: LinearFeeFunction | stutter[f] }
+}
+
+// init is a derived relation that's populated whenever an init event happens.
+fun init : Event -> LinearFeeFunction -> Int -> Int -> Int {
+  { e: Init, f: LinearFeeFunction, m: Int, s: Int, c: Int | init[f, m, s, c] }
+}
+
+// fee_bump is a derived relation that's populated whenever a fee_bump event
+// happens.
+fun fee_bump : Event -> LinearFeeFunction {
+  { e: FeeBump, f: LinearFeeFunction | fee_bump[f] }
+}
+
+// events is our final derived relation that stores all the possible event
+// types.
+fun events : set Event {
+  stutter.LinearFeeFunction + init.Int.Int.Int.LinearFeeFunction + fee_bump.LinearFeeFunction
+}
+
+// traces is a fact to ensure that we always get one of the above events in
+// traces we generate.
+fact traces {
+  always some events
+}
+
+// init_traces builds on the fact traces above, and also asserts that there's
+// always at least one init event (to ensure proper initialization).
+fact init_traces {
+  eventually (one init)
+}
+
+// fee_bump_happens ensures that within our trace, there's always at least a
+// single fee bump event.
+fact fee_bump_happens {
+  eventually (some fee_bump)
+}
+
+
+// validate_fee_rates is a fact that ensures that the ending fee rate is always
+// greater than the starting fee rate.
+fact valid_fee_rates {
+  all f: LinearFeeFunction | f.endingFeeRate > f.startingFeeRate
+}
+
+// init_correctness is an assertion that makes sure once an init event happens,
+// then we have the function in our set of active functions.
+assert init_correctness {
+  all f: LinearFeeFunction | {
+    eventually (some init) implies 
+      eventually f in ActiveFeeFunctions.activeFuncs
+  }
+}
+
+// check init_correctness
+
+// init_always_happens is a trivial assertion that ensures that an init
+// eventually happens in a trace.
+assert init_always_happens {
+   eventually (some init)
+}
+
+// check init_always_happens
+
+// init_then_fee_bump asserts that within our trace, we'll always have at least
+// a single fee bump event after an init.
+assert init_then_fee_bump {
+  eventually (some init)
+  eventually (some fee_bump)
+}
+
+// check init_then_fee_bump
+
+// bump_to_completion is a predicate that can be used to force fee_bump events
+// to be emitted until we're right at our confirmation deadline.
+pred bump_to_completion {
+  always (all f: LinearFeeFunction | f.position < f.width => eventually (some fee_bump))
+}
+
+// bump_to_final_block is a predicate that can be used to force fee_bump events
+// to be emitted until we're one block before our confirmation deadline.
+pred bump_to_final_block {
+  always (all f: LinearFeeFunction | f.position < f.width.sub[1] => eventually (some fee_bump))
+}
+
+// req_num_blocks_to_conf is a predicate that can be used to restrict the
+// functions in a fact to a given confirmation target.
+pred req_num_blocks_to_conf[n: Int] {
+  all f: LinearFeeFunction | f.width = 4
+}
```

Comment:
Is this just to constrain given the integer bounds?

also worth noting that `n` is unused here. Seems like a mistake.

Reply:
Ah yeah this is a mistake, I had it hard coded for the initial attempt to get things right, but didn't update this to actually use the param.

Fixed in the next push.